web/main.go

```diff
@@ -161,15 +180,34 @@ func logHandler(h http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
+// traceHandler decorates HandlerFunc with context that contains the trace span
+// and automatically closes the span after the request is handled.
+func (s *server) traceHandler(h func(context.Context, http.ResponseWriter, *http.Request)) http.HandlerFunc {
```

Comment:
@rakyll primary reason is that I didn't realize this package existed at all.

secondary reason is that I actually tried to create this functionality myself as

    func (HandlerFunc)HandlerFunc

however for some reason when I did `req = req.WithContext(ctx); h(w,r)` myself I realized the `r.URL.Query` was disappearing on the shallow copy.